pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller_test.go

```diff
@@ -606,8 +654,15 @@ func TestNSXServiceAccountReconciler_updateNSXServiceAccountStatus(t *testing.T)
 						ResourceVersion: "2",
 					},
 					Status: nsxvmwarecomv1alpha1.NSXServiceAccountStatus{
-						Phase:          nsxvmwarecomv1alpha1.NSXServiceAccountPhaseFailed,
-						Reason:         "Error: test error",
+						Phase:  nsxvmwarecomv1alpha1.NSXServiceAccountPhaseFailed,
+						Reason: "Error: test error",
+						Conditions: []metav1.Condition{
+							{
+								Type:   nsxvmwarecomv1alpha1.ConditionTypeRealized,
+								Status: metav1.ConditionFalse,
+								Reason: "Error: test error",
```

Comment:
I think reason should contain a programmatic identifier indicating the reason for the condition's last transition, it should like some CONST, other than a sentense, and this should fill in message.

Reply:
Not get your point. This is a unit test for `updateNSXServiceAccountStatus` func. The input error is `"test error"` and the output reason is `"Error: test error"` and the func works as expected.
Currently we don't have anything to fill `Message` section thus we keep a similar implementation as `Phase`/`Reason`, unless we want to use `Message` section.

Reply:
I think Reason should be more like enum. Such as "RealizationSuccess", "CannotReachNSX", "UnexpectedNSXErrorResponse", ... . "Error: test error" is more like a string for Message.

Reply:
OK. I think we can only put the module/step which get the error, but cannot report NSX error type since it is useless to collect the full list of NSX error.